src/herald/start_user_internet_available.py

```diff
@@ -0,0 +1,48 @@
+# start_internet_available_user.py
+#
+# Copyright (C) 2019 Kano Computing Ltd.
+# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
+#
+# Main program for start-user-internet-available script.
+
+
+import os
+
+from herald.utils import get_logged_in_users, start_user_target
+
+
+def start_internet_available_user_target(user):
+    """Start the internet-available-user.target for a given user."""
+
+    return start_user_target('internet-available-user.target', user)
+
+
+def start_internet_available_user_target_for_all_users():
+    """
+    Start the internet-available-user.target for all currently
+    logged in users.
+    """
+
+    users = get_logged_in_users()
+    users.discard('root')
+
+    successful = True
+
+    for user in users:
+        rc = start_internet_available_user_target(user)
```

Comment:
Should a user log in after this is run, how does this system work to bring the user target online for them?

Reply:
You're absolutely right, at the moment it won't do that. I suspect the solution would involve adding another target here `user-logged-in.target` which would trigger a `check-internet-connection.service` that would raise the `internet-available` targets. Will create a ticket for that.